js/CobrowseIO.js

```diff
@@ -4,6 +4,112 @@ const NativeEventEmitter = require('react-native').NativeEventEmitter
 
 const emitter = new NativeEventEmitter(CobrowseIONative)
 
+class Session {
+  constructor (session) {
+    this._session = session || {}
+    this._listen()
+  }
+
+  _update (session, onEnd) {
+    if (this._session.id === undefined || (session && this._session.id === session.id)) {
+      this._session = session
+      if (typeof onEnd === 'function') onEnd()
+    }
+  }
+
+  _listen () {
+    const requests = emitter.addListener(CobrowseIONative.SESSION_REQUESTED, (session) => this._update(session))
```

Comment:
I don't think the session object actually needs to know about the requested events (they're a oddity of the RN APIs).

Reply:
I see what you mean now. I'll remove it.